doc/developer/design/20220203_connectors.md

```diff
@@ -0,0 +1,211 @@
+    password = 'password'
+);
+```
+
+These two source definition differ only in their topic specification, but
+must duplicate eight other configuration parameters.
+
+With the connectors proposed in this document, the `CREATE SOURCE` can instead
+share all the relevant configuration:
+
+```sql
+CREATE CONNECTOR kafka FOR
+KAFKA BROKER 'kafka:9092' WITH (
+    security_protocol = 'SASL_SSL',
+    sasl_mechanisms = 'PLAIN',
+    sasl_username = 'username',
+    sasl_password = 'password'
+);
+
+CREATE CONNECTOR schema_registry FOR
+CONFLUENT SCHEMA REGISTRY 'https://schema-registry' WITH (
+    username = 'username',
+    password = 'password'
+);
+
+CREATE SOURCE kafka1
+FROM KAFKA CONNECTOR kafka TOPIC 'top1'
+FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTOR schema_registry;
+
+CREATE SOURCE kafka2
+FROM KAFKA CONNECTOR kafka TOPIC 'top2'
+FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTOR schema_registry;
+```
+
+## Design
+
+### SQL syntax and semantics
+
+Connector create syntax will come in type specific variants and will not include `with_options`, instead completely specifying the potential options in each variant. 
+
+Using connectors syntactically means replacing the source specific keyword such as BROKER or CONNECTION with CONNECTOR and supplying one or more connector names. 
+
+Properties will be merged by applying values in the supplied order with the last write winning, this supports a usage pattern of supplying base values and then progressively more specific overrides while remaining easy to reason about.  
+
+As long as connectors do not support ALTER then it is strictly not necessary to store the relationship between source and connector in the catalog, instead during planning of the CREATE SOURCE statement the values from the connector can be substituted into relevant places in the `CreateSourceConnector` without requiring any parsing changes for any part of the `CreateSourceStatement`. 
+
+While technically supporting ALTER does not have to require changing the catalog representation of sources, the utility of supporting ALTER without CASCADE is very small and supporting CASCADE requires that relations are discoverable.
+
+## Staged Implementation
+Since there is significant user experience value in the base feature without ALTER, not being able to ALTER is not a regression since sources are already immutable, and implementation is significantly less complex if connectors are only used during initial planning of a `CREATE SOURCE` statement I propose we break things into several distinct stages:
+
+1. Connectors as syntactic sugar during `CREATE SOURCE`
+2. Refactor Sources in the catalog such that connectors are referenced by relation (either implicitly or explicitly)
+    - unblock `ALTER ... CASCADE` behavior in the planner 
+    - unlocks other desirable features as well
+3. Determine exact semantics around `ALTER` and running sources once Platform has decoupled Sources into the STORAGE layer
+    - Upsert command?
```

Comment:
Agreed, later I mention that `ALTER` without `CASCADE` is effectively just syntactic sugar for `drop connector foo; create connector foo ..`. This might be desirable if we decide that `CASCADE` is too much effort for too little reward, but I doubt it.